src/tests/ftest/soak/ci_integration.py

```diff
@@ -0,0 +1,166 @@
+       uuid's to run the java-hadoop integration tests
+       using maven.
+
+    :avocado: recursive
+    """
+    def __init__(self, *args, **kwargs):
+        """Initialize a JavaCIIntegration object."""
+        super(JavaCIIntegration, self).__init__(*args, **kwargs)
+        self.pool = None
+        self.container = None
+
+    def _create_pool(self):
+        """Create a TestPool object to use with ior.
+        """
+        # Get the pool params
+        pool = TestPool(
+            self.context, dmg_command=self.get_dmg_command())
+        pool.get_params(self)
+        # Create a pool
+        pool.create()
+        return pool
+
+    def _create_cont(self, pool, path=None):
+        """Create a TestContainer object to be used to create container.
+           Args:
+               pool (TestPool): pool object
+               path (str): Unified namespace path for container
+        """
+        # Get container params
+        container = TestContainer(pool, daos_command=DaosCommand(self.bin))
+        container.get_params(self)
+        if path is not None:
+            container.path.update(path)
+        # create container
+        container.create()
+        return container
+
+    def java_version(self):
+        """Check if java is installed.
+
+        Returns:
+            bool: whether java is installed or not.
+
+        """
+
+        # checking java install
+        try:
+            result = subprocess.check_output(['java', '-version'],
+                                             stderr=subprocess.STDOUT)
+            self.log.info(result)
+        except subprocess.CalledProcessError as err:
+            self.fail("Java not installed \n {}".format(err))
+        # looking for a string something like this 1.8.0_262-b10
+        pattern = r"(\d+\.\d+\.\d+\_\d+\-[a-b]\d+)"
+
+        # replacing '-' and '-' with '.' and returning the result
+        return re.search(pattern,
+                         result).groups()[0].replace("_", ".").replace("-", ".")
+
+    def test_java_hadoop_it(self):
+        """Jira ID: DAOS-4093
+
+        Test Description:
+            Create a DAOS pool and container and use their
+            uuid's to run the java-hadoop integration tests
+            using maven.
+
+        :avocado: tags=all,pr,hw,small,javaciintegration
+        """
+        # create pool and container
+        self.pool = self._create_pool()
+        self.container = self._create_cont(self.pool)
+
+        pool_uuid = self.pool.uuid
+        cont_uuid = self.container.uuid
+
+        # obtain java version
+        version = self.java_version()
+
+        # generate jar files
+#        jdir = "{}/lib64/java".format(self.prefix)
+#        cmd = "cd {};".format(jdir)
+#        cmd += "mvn -X clean install -DskipITs -Ddaos.install.path={}".\
+#                format(self.prefix)
+#        cmd += " >> {}/maven_jar_generation_output.log".format(self.log_dir)
+#        self.execute_cmd(cmd, 180)
+
+
```

Comment:
No need for this now, yes?
```suggestion
```

Reply:
Just left it for initial iterations of testing. Will clean it all once I have a running test with proper infrastructure ready to use.